controllers/review.controller.js

```diff
@@ -29,27 +30,54 @@ class ReviewController {
         return this.formatReview(review);
     }
 
-    create(author, movieId, review) {
-        return this.reviewRepository.create({
+    async updateMovieRating(movieId) {
+        const reviews = await this.getAllByMovieId(movieId);
+        const weight = 1 / reviews.length;
+        const stats = reviews.reduce((statistics, r) => {
+            statistics.count += 1;
+            statistics.average += r.rating * weight;
+            statistics.distribution[r.rating - 1] += 1;
+            return statistics;
+        }, {
+            distribution: [0, 0, 0, 0, 0],
+            count: 0,
+            average: 0
+        });
+
+        await this.movieStatisticsRepository.findOneAndUpdate({ movieId }, {
+            movieId: movieId,
+            rating: stats
+        }, {
+            new: true,
+            upsert: true
+        });
+    }
+
+    async create(author, movieId, review) {
+        await this.reviewRepository.create({
             movieId: movieId,
             author: {
                 id: author.id,
                 name: `${author.firstname} ${author.lastname}`
```

Comment:
Why is there "{}'

Reply:
This is string interpolation, where we can insert variables into a string.